File: Problems/1910. Remove All Occurrences of a Substring/Solution.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    string removeOccurrences(string s, string part) {
        int part_len = part.length();
        string ans;

        for (char c : s) {
            ans.push_back(c);

            if (ans.length() >= part_len &&
                ans.compare(ans.length() - part_len, part_len, part) == 0) {
                ans.erase(ans.end() - part_len, ans.end());
            }
        }

        return ans;
    }
};

File: Problems/1910. Remove All Occurrences of a Substring/Solution.rs
impl Solution {
    pub fn remove_occurrences(s: String, part: String) -> String {
        let mut ans:Vec<char> = Vec::new();
        let part_chars:Vec<char> = part.chars().collect();
        let part_len = part_chars.len();

        for c in s.chars() {
            ans.push(c);

            if ans.len() >= part_len &&
                ans[ans.len() - part_len..] == part_chars[..] {
                ans.truncate(ans.len() - part_len);
            }
        }

        return ans.iter().collect();
    }
}